### PR TITLE
docs(readme): add typescript example

### PR DESCRIPTION
**Pull request by trim21**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Uzlopak** [comment]
@Fdawgs Can you review the language?

File: README.md [hunk 1]
```diff
@@ -37,6 +37,19 @@ const CustomError = createError('ERROR_CODE', 'Hello %s')
 console.log(new CustomError('world')) // error.message => 'Hello world'
 ```
 
+### TypeScript
+
+It's possible to limit your error constructor with TypeScript generic type:
```
Comment (Fdawgs) [inline README.md:42]:
```suggestion
It is possible to limit your error constructor with a generic type using TypeScript:
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by Fdawgs — APPROVED** [review]

**Comment by trim21** [comment]
Can we have a new npm release?

**Comment by climba03003** [comment]
Released

**Pull request merged**